### DISCUSSION TITLE
Generate a Open api style doc from tests that logs all api calls and their response

### DISCUSSION THREAD

**Discussion by sbland** [mswjs/msw#1451] [Ideas]
### Scope

Adds a new behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

It would be great if we could get a kind of log output of all the calls made to msw during a test run. This could then be processed to run tests on the real API to identify any issues.
Does anything exist currently that could be adapted?

**Comment by kettanaito** [comment]
Hey, @sbland. Thanks for the proposal!

You can utilize an existing [Life-cycle events API](https://mswjs.io/docs/extensions/life-cycle-events) to listen to occurring requests and serialize them to whichever format you need.

We are unlikely to ship something like this as a part of MSW. This can be an interesting tool as an ecosystem package but I think that HAR makes a much better middle ground here:

```
MSW -> HAR -> anything (e.g. OpenAPI)
```

What I'd definitely recommend is not to focus on library-specific abstractions and try to utilize standards wherever applicable.

**Reply by sbland** [reply]
Great thanks I'll take a look!
